.github/workflows/check-versions.yaml

```
@@ -36,6 +36,7 @@ jobs:
         uses: actions/setup-go@v4.1.0
         with:
           go-version: '1.21'
+          cache: false
```

```suggestion
          cache-dependency-path: ./tests
```

This would fix it for real, and still keep the caching to speed up builds.

The setup-go action looks for go.sum in the root of the repo by default. In our case the `go.mod` and `go.sum` files are located in the `tests` folder.